Source/Core/Core/PowerPC/Jit64/Jit_Integer.cpp

```diff
@@ -926,39 +928,45 @@ void Jit64::subfx(UGeckoInstruction inst)
   INSTRUCTION_START
   JITDISABLE(bJITIntegerOff);
   int a = inst.RA, b = inst.RB, d = inst.RD;
+  auto ra = regs.gpr.Lock(a);
+  auto rb = regs.gpr.Lock(b);
+  auto rd = regs.gpr.Lock(d);
 
-  if (gpr.R(a).IsImm() && gpr.R(b).IsImm())
+  if (rb.IsImm() && ra.IsImm())
   {
-    s32 i = gpr.R(b).SImm32(), j = gpr.R(a).SImm32();
-    gpr.SetImmediate32(d, i - j);
+    s32 i = rb.SImm32(), j = ra.SImm32();
+    rd.SetImm32(i - j);
     if (inst.OE)
       GenerateConstantOverflow((s64)i - (s64)j);
   }
   else
   {
-    gpr.Lock(a, b, d);
-    gpr.BindToRegister(d, (d == a || d == b), true);
-    if (d == b)
+    auto xd = rd.BindWriteAndReadIf((d == a || d == b));
+    if (d == b && d == a)
```

Comment:
Yep - it's currently an error to use the unmaterialized form of a register if you've bound it. I went back and forth on this while writing the new code because it there are a few cases where you have to add special casing like this, but it forces you to ensure that you've handled the cases where aliasing might take place.